src/bson.ts

```diff
@@ -69,3 +69,32 @@ export function pluckBSONSerializeOptions(options: BSONSerializeOptions): BSONSe
     raw
   };
 }
+
+/**
+ * Merge the given BSONSerializeOptions, preferring options over parentOptions, and optionally
+ * substituting defaults for values not set.
+ *
+ * @internal
+ *
+ * @param options - The primary options to use
+ * @param parentOptions - The secondary options to use
+ * @param includeDefaultOptions - Whether to include default options for any values not set
+ */
+export function inheritBSONOptions(
```

Comment:
small nit: we tend to call these types of methods `resolveXXX` 

Additionally, if we standardized on all "parent-like" types having a optional `bsonOptions` property (like other inheritable properties) then I think we could avoid having to pluck anything from the parent. You would just pluck the options from the user-provided options on the operation, and resolve it with a possible set of `bsonOptions` on the parent, which could be inherited up the chain back to `MongoClient`. Take a look at how `readPreference` works for a better idea of what I'm talking about here.

Reply:
This makes sense, and I think it may be along the same lines as what @reggi was suggesting earlier, since it establishes a consistent place to look for `bsonOptions` across the parent-like types and `OperationBase` types. 

One small thing: I don't think the property would be optional, since we always have defaults to fall back on. Otherwise, I see how this is better!

Reply:
agreed about defaults, once we determined there were always values then this should not be optional.